cmd/varint/varint.go

```diff
@@ -0,0 +1,85 @@
+package main
+
+import (
+	"encoding/binary"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"strconv"
+)
+
+var signed bool
+
+func main() {
+	log.SetFlags(0)
+	log.SetPrefix("varint: ")
+
+	flag.BoolVar(&signed, "s", false, "signed")
+	flagNotBoolVar(&signed, "u", true, "unsigned (default)")
+	flag.Parse()
+
+	args := flag.Args()
+	if len(args) == 0 {
+		// decode from stdin
+		b, err := ioutil.ReadAll(os.Stdin)
+		if err != nil {
+			errorf("could not read from stdin: %s", err)
+		}
+
+		var nbytes int
+		if signed {
+			var n int64
+			n, nbytes = binary.Varint(b)
+			if nbytes <= 0 {
+				errorf("could not parse varint")
+			}
+			_, err = os.Stdout.Write([]byte(strconv.FormatInt(n, 10)))
+			if err != nil {
+				errorf("could not write to stdout: %s", err)
+			}
+		} else {
+			var n uint64
+			n, nbytes = binary.Uvarint(b)
+			if nbytes <= 0 {
+				errorf("could not parse uvarint")
+			}
+			_, err = os.Stdout.Write([]byte(strconv.FormatUint(n, 10)))
+			if err != nil {
+				errorf("could not write to stdout: %s", err)
+			}
+		}
+		return
+	}
+
+	// encode from args
+	if len(args) != 1 {
+		errorf("invalid argument count %d; varint must read from stdin or take 1 argument", len(args))
+	}
+
+	val, err := strconv.ParseInt(args[0], 10, 64)
```

Comment:
We might need to use ParseUint for the unsigned case. For example if the command is `varint -u 9223372036854775808` then ParseInt will return an error. (Not a big deal because we probably won't be exercising those edge cases in practice, but it'd be best to handle it correctly, heh.)